### ISSUE TITLE
TF lite implementation for RandomStandardNormal

### ISSUE THREAD

**Issue by DocDriven** [tensorflow/tensorflow#33341]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): Docker image tf-nightly-gpu-py3
- TensorFlow version (or github SHA if from source): 3.6.8

**Provide the text output from tflite_convert**

```
Some of the operators in the model are not supported by the standard TensorFlow Lite runtime. If those are native TensorFlow operators, you might be able to use the extended runtime by passing --enable_select_tf_ops, or by setting target_ops=TFLITE_BUILTINS,SELECT_TF_OPS when calling tf.lite.TFLiteConverter(). Otherwise, if you have a custom implementation for them you can disable this error with --allow_custom_ops, or by setting allow_custom_ops=True when calling tf.lite.TFLiteConverter(). Here is a list of builtin operators you are using: ADD, EXP, FULLY_CONNECTED, LEAKY_RELU, LOG, MUL. Here is a list of operators for which you will need custom implementations: RandomStandardNormal.
```

VAE model was taken from your custom layers and models site ([here](https://www.tensorflow.org/guide/keras/custom_layers_and_models#building_models)). The line of code producing this exception is
`epsilon = tf.keras.backend.random_normal(shape=(batch, dim))`.

It might be a good idea to add a tflite implementation for this function so that people can try a conversion with your existing examples.

**Comment by DocDriven** [comment]
@jaeyoo Do you have any updates regarding the implementation? There seems to be some work in progress for the related issue #34277 [here](https://github.com/orgs/tensorflow/projects/8#card-32750656), but not for the actual implementation. 

If you have any experimental implementation for testing, please let me know.

**Comment by jaeyoo** [comment]
Hi, @DocDriven,

I could convert the model in tf 2.2.0 with the select_ops
```python
converter = tf.lite.TFLiteConverter.from_keras_model(vae)
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS,
                                       tf.lite.OpsSet.SELECT_TF_OPS]
tflite_model = converter.convert()
```
Could you try it?

**Comment by CoreyCole** [comment]
Hey @jaeyoo I am able to convert my model using SELECT_TF_OPS, but that greatly increases the size of the libtensorflowlite library.

I am trying to get my model running on Android and this is the one operator in my way from >100 megabytes decrease in app download size.

Not only that, but I am trying to use the tensorflowlite c api and it doesn't seem like android arm compilation is supported when using flex delegate:

```
ERROR: /Users/corey/Workspace/omnivor-io/gst-sync-android/app/src/main/jni/tensorflow/tensorflow/lite/delegates/flex/BUILD:76:1: C++ compilation of rule '//tensorflow/lite/delegates/flex:delegate_only_runtime' failed (Exit 1): clang failed: error executing command 
  (cd /private/var/tmp/_bazel_corey/9e892e94b2950203c06bbbfc8119bc78/execroot/org_tensorflow && \
  exec env - \
    ANDROID_BUILD_TOOLS_VERSION=29.0.3 \
    ANDROID_NDK_API_LEVEL=26 \
    ANDROID_NDK_HOME=/Users/corey/library/Android/Sdk/ndk-bundle \
    ANDROID_SDK_API_LEVEL=29 \
    ANDROID_SDK_HOME=/Users/corey/library/Android/Sdk \
    PATH=/Users/corey/.pub-cache/bin:/Users/corey/Workspace/depot_tools:/usr/local/opt/llvm/bin:/Users/corey/Downloads/google-cloud-sdk/bin:/Users/corey/.rbenv/shims:/Users/corey/Library/Android/sdk/platform-tools:/Users/corey/Library/Android/sdk/tools:/Users/corey/.fastlane/bin:/Users/corey/.local/bin:/Library/Frameworks/GStreamer.framework/Versions/Current/bin/:/Users/corey/.poetry/bin:/Users/corey/Library/Android/sdk/tools/bin/:/Applications/Postgres.app/Contents/Versions/10/bin:/Users/corey/.pyenv/shims:/usr/local/bin:/Users/corey/local/bin:/Users/corey/.npm-global/node/bin:/Users/corey/.npm-global/bin:/Users/corey/.poetry/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/corey/.ebcli-virtual-env/executables:/Users/corey/.pub-cache/bin:/Users/corey/Workspace/depot_tools:/usr/local/opt/llvm/bin:/Users/corey/Downloads/google-cloud-sdk/bin:/Users/corey/.rbenv/shims:/Users/corey/Library/Android/sdk/platform-tools:/Users/corey/Library/Android/sdk/tools:/Users/corey/.fastlane/bin:/Users/corey/.local/bin:/Library/Frameworks/GStreamer.framework/Versions/Current/bin/:/Users/corey/.poetry/bin:/Users/corey/Library/Android/sdk/tools/bin/:/Applications/Postgres.app/Contents/Versions/10/bin:/Users/corey/.pyenv/shims:/Users/corey/local/bin:/Users/corey/.npm-global/node/bin:/Users/corey/.npm-global/bin:/Users/corey/flutter/bin:/Users/corey/bin:/Users/corey/flutter/bin:/Users/corey/bin \
    PWD=/proc/self/cwd \
    PYTHON_BIN_PATH=/Users/corey/.pyenv/versions/3.7.3/bin/python3 \
    PYTHON_LIB_PATH=/Users/corey/.pyenv/versions/3.7.3/lib/python3.7/site-packages \
    TF2_BEHAVIOR=1 \
    TF_CONFIGURE_IOS=0 \
    TF_ENABLE_XLA=1 \
  external/androidndk/ndk/toolchains/llvm/prebuilt/darwin-x86_64/bin/clang -gcc-toolchain external/androidndk/ndk/toolchains/aarch64-linux-android-4.9/prebuilt/darwin-x86_64 -target aarch64-none-linux-android -fpic -isystemexternal/androidndk/ndk/sysroot/usr/include/aarch64-linux-android '-D__ANDROID_API__=26' -no-canonical-prefixes -Wno-invalid-command-line-argument -Wno-unused-command-line-argument -funwind-tables -fstack-protector-strong -fno-addrsig '-Werror=return-type' '-Werror=int-to-pointer-cast' '-Werror=pointer-to-int-cast' '-Werror=implicit-function-declaration' -O2 -g -DNDEBUG -MD -MF bazel-out/arm64-v8a-opt/bin/tensorflow/lite/delegates/flex/_objs/delegate_only_runtime/delegate.pic.d '-frandom-seed=bazel-out/arm64-v8a-opt/bin/tensorflow/lite/delegates/flex/_objs/delegate_only_runtime/delegate.pic.o' -fPIC -DEIGEN_MPL2_ONLY '-DEIGEN_MAX_ALIGN_BYTES=64' '-DEIGEN_HAS_TYPE_TRAITS=0' -DSUPPORT_SELECTIVE_REGISTRATION -D__CLANG_SUPPORT_DYN_ANNOTATION__ -iquote . -iquote bazel-out/arm64-v8a-opt/bin -iquote external/com_google_absl -iquote bazel-out/arm64-v8a-opt/bin/external/com_google_absl -iquote external/nsync -iquote bazel-out/arm64-v8a-opt/bin/external/nsync -iquote external/eigen_archive -iquote bazel-out/arm64-v8a-opt/bin/external/eigen_archive -iquote external/local_config_sycl -iquote bazel-out/arm64-v8a-opt/bin/external/local_config_sycl -iquote external/com_google_protobuf -iquote bazel-out/arm64-v8a-opt/bin/external/com_google_protobuf -iquote external/zlib -iquote bazel-out/arm64-v8a-opt/bin/external/zlib -iquote external/double_conversion -iquote bazel-out/arm64-v8a-opt/bin/external/double_conversion -iquote external/com_googlesource_code_re2 -iquote bazel-out/arm64-v8a-opt/bin/external/com_googlesource_code_re2 -iquote external/farmhash_archive -iquote bazel-out/arm64-v8a-opt/bin/external/farmhash_archive -iquote external/flatbuffers -iquote bazel-out/arm64-v8a-opt/bin/external/flatbuffers -Ibazel-out/arm64-v8a-opt/bin/external/flatbuffers/_virtual_includes/flatbuffers -Ibazel-out/arm64-v8a-opt/bin/external/flatbuffers/src/_virtual_includes/flatbuffers -Ibazel-out/arm64-v8a-opt/bin/external/flatbuffers/_virtual_includes/runtime_cc -isystem external/nsync/public -isystem bazel-out/arm64-v8a-opt/bin/external/nsync/public -isystem external/eigen_archive -isystem bazel-out/arm64-v8a-opt/bin/external/eigen_archive -isystem external/com_google_protobuf/src -isystem bazel-out/arm64-v8a-opt/bin/external/com_google_protobuf/src -isystem external/zlib -isystem bazel-out/arm64-v8a-opt/bin/external/zlib -isystem external/double_conversion -isystem bazel-out/arm64-v8a-opt/bin/external/double_conversion -isystem external/farmhash_archive/src -isystem bazel-out/arm64-v8a-opt/bin/external/farmhash_archive/src -isystem tensorflow/lite/schema -isystem bazel-out/arm64-v8a-opt/bin/tensorflow/lite/schema -w '-std=c++14' '--sysroot=external/androidndk/ndk/platforms/android-26/arch-arm64' -isystem external/androidndk/ndk/sources/cxx-stl/llvm-libc++/include -isystem external/androidndk/ndk/sources/cxx-stl/llvm-libc++abi/include -isystem external/androidndk/ndk/sources/android/support/include -isystemexternal/androidndk/ndk/sysroot/usr/include -c tensorflow/lite/delegates/flex/delegate.cc -o bazel-out/arm64-v8a-opt/bin/tensorflow/lite/delegates/flex/_objs/delegate_only_runtime/delegate.pic.o)
Execution platform: @local_execution_config_platform//:platform
tensorflow/lite/delegates/flex/delegate.cc:143:44: error: 'TF_AcquireFlexDelegate' has C-linkage specified, but returns user-defined type 'tflite::TfLiteDelegateUniquePtr' (aka 'unique_ptr<TfLiteDelegate, void (*)(TfLiteDelegate *)>') which is incompatible with C [-Werror,-Wreturn-type-c-linkage]
extern "C" tflite::TfLiteDelegateUniquePtr TF_AcquireFlexDelegate() {
                                           ^
1 error generated.
Target //tensorflow/lite/c:libtensorflowlite_c.so failed to build
```

So for my use case, I believe I need RandomStandardNormal without flex delegate / SELECT_TF_OPS.

**Comment by CoreyCole** [comment]
I noticed a commit in `v2.4.0-rc0` that has create a custom prototype for random standard normal:

https://github.com/tensorflow/tensorflow/commit/11823c6179a5e2ca6ac4f8480aa00c42c772885e

I've compiled this library but I'm getting the error:

```
2020-11-04 17:52:41.409 25623-25623/io.kernellabs.sync E/tflite: Encountered unresolved custom op: RandomStandardNormal.
2020-11-04 17:52:41.409 25623-25623/io.kernellabs.sync E/tflite: Node number 1 (RandomStandardNormal) failed to prepare.
```

Any guidance on how I can register the op using the c api?

**Comment by henry8th** [comment]
@CoreyCole Hello, Did you manage to find a solution?

**Comment by CoreyCole** [comment]
@henry8th yes, hope this helps: https://github.com/tensorflow/tensorflow/issues/44664#issuecomment-723310060

**Comment by henry8th** [comment]
Thank you

**Comment by saikumarchalla** [comment]
@DocDriven  Is this still an issue for you? If not, Please go ahead and close the issue.Thanks!
